agent/src/agentclient.rs

```diff
@@ -227,16 +225,16 @@ impl BrupopAgent {
                         // and then we need to wait until the host is successfully drained before transitioning
                         // to the next state.
                     }
-                    BottlerocketNodeState::PerformingUpdate => {
+                    BottlerocketNodeState::PerformedUpdate => {
                         log::info!("Performing update");
                         update().await.context(error::UpdateActions {
                             action: "Perform".to_string(),
                         })?;
                     }
-                    BottlerocketNodeState::RebootingToUpdate => {
+                    BottlerocketNodeState::RebootedToUpdate => {
                         log::info!("Rebooting node to complete update");
 
-                        if ensure_reboot_happened(&bottlerocket_node_status)? {
+                        if running_desired_version(&bottlerocket_node_spec).await? {
```

Comment:
Yes, it can. It's not yet handled, but the plan is to track the number of attempted updates, eventually giving up and emitting a failure metric to Prometheus via opentelemetry. That handling wasn't codified in an issue, so I've opened #123